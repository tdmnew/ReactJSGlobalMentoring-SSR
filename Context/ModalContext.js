import React from "react";

const useModalState = (initialState) => {
    const [isOpen, setIsOpen] = React.useState(initialState.isOpen);
    const [modalProps, setModalProps] = React.useState(initialState.modalProps);

    const setModalState = ({ isOpen, modalProps = {} }) => {
        setIsOpen(isOpen);
        setModalProps(modalProps);
    };

    return [{ isOpen, modalProps }, setModalState];
};

export const ModalStateContext = React.createContext();
export const ModalUpdaterContext = React.createContext();

export const ModalContext = ({ children }) => {
    const [modalOptions, setModalOptions] = useModalState({
        isOpen: false,
        modalProps: {},
    });

    return (
            <ModalUpdaterContext.Provider value={setModalOptions}>
                <ModalStateContext.Provider value={modalOptions}>
                    {children}
                </ModalStateContext.Provider>
            </ModalUpdaterContext.Provider>
    );
};
