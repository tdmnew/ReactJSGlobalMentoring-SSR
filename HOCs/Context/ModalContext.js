import React, { createContext, useState } from 'react';

const useModalState = (initialState) => {
  const [isOpen, setIsOpen] = useState(initialState.isOpen);
  const [modalProps, setModalProps] = useState(initialState.modalProps);

  const setModalState = ({ isOpen, modalProps = {} }) => {
    setIsOpen(isOpen);
    setModalProps(modalProps);
  };

  return [{ isOpen, modalProps }, setModalState];
};

export const ModalStateContext = createContext();
export const ModalUpdaterContext = createContext();

export const ModalContext = ({ children }) => {
  const [modalOptions, setModalOptions] = useModalState({
    isOpen: false,
    modalProps: {},
  });

  return (
    <ModalUpdaterContext.Provider value={setModalOptions}>
      <ModalStateContext.Provider value={modalOptions}>
        {children}
      </ModalStateContext.Provider>
    </ModalUpdaterContext.Provider>
  );
};
