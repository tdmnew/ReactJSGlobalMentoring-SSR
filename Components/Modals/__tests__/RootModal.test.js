)import React from "react";

import { render, screen, cleanup, waitFor } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";

import RootModal from "../RootModal";

import { ModalStateContext } from "../../../HOCs/Context/ModalContext";

describe("Root Modal", () => {
    afterEach(cleanup);

    const initialState = {
        movies: [],
        selectedMovie: {},
    };

    const mockStore = configureStore();
    const store = mockStore(initialState);

    const addMovieOptions = {
        isOpen: true,
        modalProps: { type: "Add Movie" },
    };

    const editMovieOptions = {
        isOpen: true,
        modalProps: {
            type: "Edit Movie",
            info: {
                id: 1,
                title: "Blade Runner",
                tagline: "",
                vote_average: 4.3,
                vote_count: 100,
                genres: ["Sci-Fi"],
                release_date: "1982-01-01",
                overview: "",
                budget: 100000,
                revenue: 200000,
                runtime: 120,
                poster_path:
                    "https://d13ezvd6yrslxm.cloudfront.net/wp/wp-content/images/BR-titled.jpg",
            },
        },
    };

    const deleteMovieOptions = {
        isOpen: true,
        modalProps: {
            type: "Delete Movie",
            info: {
                id: 1,
            },
        },
    };

    it("Changes input on Add Movie fields", () => {
        const addMovie = render(
            <Provider store={store}>
                <ModalStateContext.Provider value={addMovieOptions}>
                    <RootModal />
                </ModalStateContext.Provider>
            </Provider>
        );

        const title = screen.getByLabelText(/title/i);
        userEvent.type(title, "test");
        expect(title).toHaveValue("test");

        const release = screen.getByPlaceholderText("Movie Release Date");
        userEvent.type(release, "1990-01-01");
        expect(release).toHaveValue("1990-01-01");

        const url = screen.getByLabelText(/url/i);
        userEvent.type(url, "test");
        expect(url).toHaveValue("test");

        const overview = screen.getByLabelText(/overview/i);
        userEvent.type(overview, "test");
        expect(overview).toHaveValue("test");

        const runtime = screen.getByLabelText(/runtime/i);
        userEvent.type(runtime, "120".toString());
        expect(runtime).toHaveValue(120);
    });

    it("Changes input on Edit Movie fields", () => {
        const editMovie = render(
            <Provider store={store}>
                <ModalStateContext.Provider value={editMovieOptions}>
                    <RootModal />
                </ModalStateContext.Provider>
            </Provider>
        );

        const title = screen.getByLabelText(/title/i);
        userEvent.clear(title);
        userEvent.type(title, "test");
        expect(title).toHaveValue("test");

        const release = screen.getByPlaceholderText("Movie Release Date");
        userEvent.clear(release);
        userEvent.type(release, "1990-01-01");
        expect(release).toHaveValue("1990-01-01");

        const url = screen.getByLabelText(/url/i);
        userEvent.clear(url);
        userEvent.type(url, "test");
        expect(url).toHaveValue("test");

        const overview = screen.getByLabelText(/overview/i);
        userEvent.clear(overview);
        userEvent.type(overview, "test");
        expect(overview).toHaveValue("test");

        const runtime = screen.getByLabelText(/runtime/i);
        userEvent.clear(runtime);
        userEvent.type(runtime, "120".toString());
        expect(runtime).toHaveValue(120);
    });

    it("Shows validation errors on Add Movie", async () => {
        const addMovie = render(
            <Provider store={store}>
                <ModalStateContext.Provider value={addMovieOptions}>
                    <RootModal />
                </ModalStateContext.Provider>
            </Provider>
        );

        const submit = screen.getByTitle("submit");

        await waitFor(() => userEvent.click(submit));
        expect(screen.getByText(/Title Required/i)).toBeInTheDocument();
        expect(screen.getByText(/Release Date Required/i)).toBeInTheDocument();
        expect(
            screen.getByText(/Please select at least one genre/i)
        ).toBeInTheDocument();
        expect(screen.getByText(/Overview Required/i)).toBeInTheDocument();
        expect(screen.getByText(/Runtime Required/i)).toBeInTheDocument();
    });

    it("Shows validation errors on Edit Movie", async () => {
        const editMovie = render(
            <Provider store={store}>
                <ModalStateContext.Provider value={editMovieOptions}>
                    <RootModal />
                </ModalStateContext.Provider>
            </Provider>
        );
        const title = screen.getByLabelText(/title/i);
        userEvent.clear(title);

        const release = screen.getByPlaceholderText("Movie Release Date");
        userEvent.clear(release);

        const url = screen.getByLabelText(/url/i);
        userEvent.clear(url);

        const overview = screen.getByLabelText(/overview/i);
        userEvent.clear(overview);

        const runtime = screen.getByLabelText(/runtime/i);
        userEvent.clear(runtime);
        userEvent.type(runtime, "0");

        const submit = screen.getByTitle("submit");

        await waitFor(() => userEvent.click(submit));
        expect(screen.getByText(/Title Required/i)).toBeInTheDocument();
        expect(screen.getByText(/Release Date Required/i)).toBeInTheDocument();
        expect(screen.getByText(/Overview Required/i)).toBeInTheDocument();
        expect(screen.getByText(/Runtime Required/i)).toBeInTheDocument();
    });

    it("Renders the Edit Movie Modal", () => {
        const tree = render(
            <Provider store={store}>
                <ModalStateContext.Provider value={editMovieOptions}>
                    <RootModal />
                </ModalStateContext.Provider>
            </Provider>
        );

        expect(tree).toMatchSnapshot();

        expect(screen.getByLabelText(/title/i)).toBeInTheDocument();
        expect(screen.getByLabelText(/release date/i)).toBeInTheDocument();
        expect(screen.getByLabelText(/url/i)).toBeInTheDocument();
        expect(screen.getByLabelText(/overview/i)).toBeInTheDocument();
        expect(screen.getByLabelText(/runtime/i)).toBeInTheDocument();
    });

    it("Renders the Add Movie Modal", () => {
        const addMovie = render(
            <Provider store={store}>
                <ModalStateContext.Provider value={addMovieOptions}>
                    <RootModal />
                </ModalStateContext.Provider>
            </Provider>
        );

        expect(addMovie).toMatchSnapshot();

        expect(screen.getByLabelText(/title/i)).toBeInTheDocument();
        expect(screen.getByLabelText(/release date/i)).toBeInTheDocument();
        expect(screen.getByLabelText(/url/i)).toBeInTheDocument();
        expect(screen.getByLabelText(/overview/i)).toBeInTheDocument();
        expect(screen.getByLabelText(/runtime/i)).toBeInTheDocument();
    });

    it("Renders the Delete Movie Modal", () => {
        const tree = render(
            <Provider store={store}>
                <ModalStateContext.Provider value={deleteMovieOptions}>
                    <RootModal />
                </ModalStateContext.Provider>
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });
});;
