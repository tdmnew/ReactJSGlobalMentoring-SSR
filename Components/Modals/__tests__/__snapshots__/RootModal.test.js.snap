// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root Modal Renders the Add Movie Modal 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-modal"
      >
        <div
          class="form-modal container"
        >
          <button
            class="container__close-btn"
            title="close"
          >
            X
          </button>
          <form>
            <h2>
              ADD MOVIE
            </h2>
            <label
              aria-labelledby="title"
            >
              <span>
                TITLE
              </span>
              <input
                class="container__input"
                name="title"
                placeholder="Movie Title"
                title="title"
                value=""
              />
            </label>
            <label
              aria-labelledby="release date"
            >
              <span>
                RELEASE DATE
              </span>
              <input
                class="container__date"
                name="release_date"
                placeholder="Movie Release Date"
                title="release_date"
                type="date"
                value=""
              />
            </label>
            <label
              aria-labelledby="url"
            >
              <span>
                MOVIE URL
              </span>
              <input
                class="container__input"
                name="url"
                placeholder="Movie Website"
                title="url"
                value=""
              />
            </label>
            <label
              aria-labelledby="genres"
            >
              <span>
                GENRE
              </span>
              <textarea
                class="container dropdown-btn"
              />
              <div
                class="container dropdown"
                style="display: none;"
              >
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Action"
                    type="checkbox"
                    value="Action"
                  />
                  <span
                    class="option__text"
                  >
                    Action
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Adventure"
                    type="checkbox"
                    value="Adventure"
                  />
                  <span
                    class="option__text"
                  >
                    Adventure
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Animation"
                    type="checkbox"
                    value="Animation"
                  />
                  <span
                    class="option__text"
                  >
                    Animation
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Comedy"
                    type="checkbox"
                    value="Comedy"
                  />
                  <span
                    class="option__text"
                  >
                    Comedy
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Crime"
                    type="checkbox"
                    value="Crime"
                  />
                  <span
                    class="option__text"
                  >
                    Crime
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Drama"
                    type="checkbox"
                    value="Drama"
                  />
                  <span
                    class="option__text"
                  >
                    Drama
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Fantasy"
                    type="checkbox"
                    value="Fantasy"
                  />
                  <span
                    class="option__text"
                  >
                    Fantasy
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Mystery"
                    type="checkbox"
                    value="Mystery"
                  />
                  <span
                    class="option__text"
                  >
                    Mystery
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Romance"
                    type="checkbox"
                    value="Romance"
                  />
                  <span
                    class="option__text"
                  >
                    Romance
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Science Fiction"
                    type="checkbox"
                    value="Science Fiction"
                  />
                  <span
                    class="option__text"
                  >
                    Science Fiction
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Thriller"
                    type="checkbox"
                    value="Thriller"
                  />
                  <span
                    class="option__text"
                  >
                    Thriller
                  </span>
                </label>
              </div>
            </label>
            <label
              aria-labelledby="overview"
            >
              <span>
                OVERVIEW
              </span>
              <input
                class="container__input"
                name="overview"
                placeholder="Movie Overview"
                title="overview"
                value=""
              />
            </label>
            <label
              aria-labelledby="runtime"
            >
              <span>
                RUNTIME
              </span>
              <input
                class="container__input"
                min="0"
                name="runtime"
                placeholder="Movie Runtime"
                title="runtime"
                type="number"
                value="0"
              />
            </label>
            <div
              class="container buttons"
            >
              <button
                class="container buttons__reset"
                title="reset"
              >
                RESET
              </button>
              <button
                class="container buttons__submit"
                title="submit"
                type="submit"
              >
                SUBMIT
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-modal"
    >
      <div
        class="form-modal container"
      >
        <button
          class="container__close-btn"
          title="close"
        >
          X
        </button>
        <form>
          <h2>
            ADD MOVIE
          </h2>
          <label
            aria-labelledby="title"
          >
            <span>
              TITLE
            </span>
            <input
              class="container__input"
              name="title"
              placeholder="Movie Title"
              title="title"
              value=""
            />
          </label>
          <label
            aria-labelledby="release date"
          >
            <span>
              RELEASE DATE
            </span>
            <input
              class="container__date"
              name="release_date"
              placeholder="Movie Release Date"
              title="release_date"
              type="date"
              value=""
            />
          </label>
          <label
            aria-labelledby="url"
          >
            <span>
              MOVIE URL
            </span>
            <input
              class="container__input"
              name="url"
              placeholder="Movie Website"
              title="url"
              value=""
            />
          </label>
          <label
            aria-labelledby="genres"
          >
            <span>
              GENRE
            </span>
            <textarea
              class="container dropdown-btn"
            />
            <div
              class="container dropdown"
              style="display: none;"
            >
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Action"
                  type="checkbox"
                  value="Action"
                />
                <span
                  class="option__text"
                >
                  Action
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Adventure"
                  type="checkbox"
                  value="Adventure"
                />
                <span
                  class="option__text"
                >
                  Adventure
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Animation"
                  type="checkbox"
                  value="Animation"
                />
                <span
                  class="option__text"
                >
                  Animation
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Comedy"
                  type="checkbox"
                  value="Comedy"
                />
                <span
                  class="option__text"
                >
                  Comedy
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Crime"
                  type="checkbox"
                  value="Crime"
                />
                <span
                  class="option__text"
                >
                  Crime
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Drama"
                  type="checkbox"
                  value="Drama"
                />
                <span
                  class="option__text"
                >
                  Drama
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Fantasy"
                  type="checkbox"
                  value="Fantasy"
                />
                <span
                  class="option__text"
                >
                  Fantasy
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Mystery"
                  type="checkbox"
                  value="Mystery"
                />
                <span
                  class="option__text"
                >
                  Mystery
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Romance"
                  type="checkbox"
                  value="Romance"
                />
                <span
                  class="option__text"
                >
                  Romance
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Science Fiction"
                  type="checkbox"
                  value="Science Fiction"
                />
                <span
                  class="option__text"
                >
                  Science Fiction
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Thriller"
                  type="checkbox"
                  value="Thriller"
                />
                <span
                  class="option__text"
                >
                  Thriller
                </span>
              </label>
            </div>
          </label>
          <label
            aria-labelledby="overview"
          >
            <span>
              OVERVIEW
            </span>
            <input
              class="container__input"
              name="overview"
              placeholder="Movie Overview"
              title="overview"
              value=""
            />
          </label>
          <label
            aria-labelledby="runtime"
          >
            <span>
              RUNTIME
            </span>
            <input
              class="container__input"
              min="0"
              name="runtime"
              placeholder="Movie Runtime"
              title="runtime"
              type="number"
              value="0"
            />
          </label>
          <div
            class="container buttons"
          >
            <button
              class="container buttons__reset"
              title="reset"
            >
              RESET
            </button>
            <button
              class="container buttons__submit"
              title="submit"
              type="submit"
            >
              SUBMIT
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Root Modal Renders the Delete Movie Modal 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="message-modal"
      >
        <div
          class="message-modal container"
        >
          <button
            class="container__close-btn"
          >
            X
          </button>
          <form>
            <h2>
              DELETE MOVIE
            </h2>
            <p>
              Are you sure you want to delete this movie?
            </p>
            <div
              class="container buttons"
            >
              <button
                class="container buttons__confirm"
                type="submit"
              >
                CONFIRM
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="message-modal"
    >
      <div
        class="message-modal container"
      >
        <button
          class="container__close-btn"
        >
          X
        </button>
        <form>
          <h2>
            DELETE MOVIE
          </h2>
          <p>
            Are you sure you want to delete this movie?
          </p>
          <div
            class="container buttons"
          >
            <button
              class="container buttons__confirm"
              type="submit"
            >
              CONFIRM
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Root Modal Renders the Edit Movie Modal 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form-modal"
      >
        <div
          class="form-modal container"
        >
          <button
            class="container__close-btn"
            title="close"
          >
            X
          </button>
          <form>
            <h2>
              EDIT MOVIE
            </h2>
            <label
              aria-labelledby="movieid"
            >
              <span>
                MOVIE ID
              </span>
              <input
                class="container__movieid"
                disabled=""
                name="movieid"
                placeholder="Movie ID"
                value="1"
              />
            </label>
            <label
              aria-labelledby="title"
            >
              <span>
                TITLE
              </span>
              <input
                class="container__input"
                name="title"
                placeholder="Movie Title"
                value="Blade Runner"
              />
            </label>
            <label
              aria-labelledby="release_date"
            >
              <span>
                RELEASE DATE
              </span>
              <input
                class="container__date"
                name="release_date"
                placeholder="Movie Release Date"
                type="date"
                value="1982-01-01"
              />
            </label>
            <label
              aria-labelledby="url"
            >
              <span>
                MOVIE URL
              </span>
              <input
                class="container__input"
                name="url"
                placeholder="Movie Website"
                value=""
              />
            </label>
            <label
              aria-labelledby="genre"
            >
              <span>
                GENRE
              </span>
              <textarea
                class="container dropdown-btn"
              >
                Sci-Fi
              </textarea>
              <div
                class="container dropdown"
                style="display: none;"
              >
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Action"
                    type="checkbox"
                    value="Action"
                  />
                  <span
                    class="option__text"
                  >
                    Action
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Adventure"
                    type="checkbox"
                    value="Adventure"
                  />
                  <span
                    class="option__text"
                  >
                    Adventure
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Animation"
                    type="checkbox"
                    value="Animation"
                  />
                  <span
                    class="option__text"
                  >
                    Animation
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Comedy"
                    type="checkbox"
                    value="Comedy"
                  />
                  <span
                    class="option__text"
                  >
                    Comedy
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Crime"
                    type="checkbox"
                    value="Crime"
                  />
                  <span
                    class="option__text"
                  >
                    Crime
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Drama"
                    type="checkbox"
                    value="Drama"
                  />
                  <span
                    class="option__text"
                  >
                    Drama
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Fantasy"
                    type="checkbox"
                    value="Fantasy"
                  />
                  <span
                    class="option__text"
                  >
                    Fantasy
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Mystery"
                    type="checkbox"
                    value="Mystery"
                  />
                  <span
                    class="option__text"
                  >
                    Mystery
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Romance"
                    type="checkbox"
                    value="Romance"
                  />
                  <span
                    class="option__text"
                  >
                    Romance
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Science Fiction"
                    type="checkbox"
                    value="Science Fiction"
                  />
                  <span
                    class="option__text"
                  >
                    Science Fiction
                  </span>
                </label>
                <label
                  class="dropdown option"
                >
                  <input
                    class="option__checkbox"
                    name="Thriller"
                    type="checkbox"
                    value="Thriller"
                  />
                  <span
                    class="option__text"
                  >
                    Thriller
                  </span>
                </label>
              </div>
            </label>
            <label
              aria-labelledby="overview"
            >
              <span>
                OVERVIEW
              </span>
              <input
                class="container__input"
                name="overview"
                placeholder="Movie Overview"
                value=""
              />
            </label>
            <label
              aria-labelledby="runtime"
            >
              <span>
                RUNTIME
              </span>
              <input
                class="container__input"
                min="0"
                name="runtime"
                placeholder="Movie Runtime"
                type="number"
                value="120"
              />
            </label>
            <div
              class="container buttons"
            >
              <button
                class="container buttons__reset"
                title="reset"
              >
                RESET
              </button>
              <button
                class="container buttons__submit"
                title="submit"
                type="submit"
              >
                SUBMIT
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form-modal"
    >
      <div
        class="form-modal container"
      >
        <button
          class="container__close-btn"
          title="close"
        >
          X
        </button>
        <form>
          <h2>
            EDIT MOVIE
          </h2>
          <label
            aria-labelledby="movieid"
          >
            <span>
              MOVIE ID
            </span>
            <input
              class="container__movieid"
              disabled=""
              name="movieid"
              placeholder="Movie ID"
              value="1"
            />
          </label>
          <label
            aria-labelledby="title"
          >
            <span>
              TITLE
            </span>
            <input
              class="container__input"
              name="title"
              placeholder="Movie Title"
              value="Blade Runner"
            />
          </label>
          <label
            aria-labelledby="release_date"
          >
            <span>
              RELEASE DATE
            </span>
            <input
              class="container__date"
              name="release_date"
              placeholder="Movie Release Date"
              type="date"
              value="1982-01-01"
            />
          </label>
          <label
            aria-labelledby="url"
          >
            <span>
              MOVIE URL
            </span>
            <input
              class="container__input"
              name="url"
              placeholder="Movie Website"
              value=""
            />
          </label>
          <label
            aria-labelledby="genre"
          >
            <span>
              GENRE
            </span>
            <textarea
              class="container dropdown-btn"
            >
              Sci-Fi
            </textarea>
            <div
              class="container dropdown"
              style="display: none;"
            >
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Action"
                  type="checkbox"
                  value="Action"
                />
                <span
                  class="option__text"
                >
                  Action
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Adventure"
                  type="checkbox"
                  value="Adventure"
                />
                <span
                  class="option__text"
                >
                  Adventure
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Animation"
                  type="checkbox"
                  value="Animation"
                />
                <span
                  class="option__text"
                >
                  Animation
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Comedy"
                  type="checkbox"
                  value="Comedy"
                />
                <span
                  class="option__text"
                >
                  Comedy
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Crime"
                  type="checkbox"
                  value="Crime"
                />
                <span
                  class="option__text"
                >
                  Crime
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Drama"
                  type="checkbox"
                  value="Drama"
                />
                <span
                  class="option__text"
                >
                  Drama
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Fantasy"
                  type="checkbox"
                  value="Fantasy"
                />
                <span
                  class="option__text"
                >
                  Fantasy
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Mystery"
                  type="checkbox"
                  value="Mystery"
                />
                <span
                  class="option__text"
                >
                  Mystery
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Romance"
                  type="checkbox"
                  value="Romance"
                />
                <span
                  class="option__text"
                >
                  Romance
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Science Fiction"
                  type="checkbox"
                  value="Science Fiction"
                />
                <span
                  class="option__text"
                >
                  Science Fiction
                </span>
              </label>
              <label
                class="dropdown option"
              >
                <input
                  class="option__checkbox"
                  name="Thriller"
                  type="checkbox"
                  value="Thriller"
                />
                <span
                  class="option__text"
                >
                  Thriller
                </span>
              </label>
            </div>
          </label>
          <label
            aria-labelledby="overview"
          >
            <span>
              OVERVIEW
            </span>
            <input
              class="container__input"
              name="overview"
              placeholder="Movie Overview"
              value=""
            />
          </label>
          <label
            aria-labelledby="runtime"
          >
            <span>
              RUNTIME
            </span>
            <input
              class="container__input"
              min="0"
              name="runtime"
              placeholder="Movie Runtime"
              type="number"
              value="120"
            />
          </label>
          <div
            class="container buttons"
          >
            <button
              class="container buttons__reset"
              title="reset"
            >
              RESET
            </button>
            <button
              class="container buttons__submit"
              title="submit"
              type="submit"
            >
              SUBMIT
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
